client.go

```diff
@@ -164,27 +165,30 @@ func NewClient(config *Config, remote string) (*Client, error) {
 			info.RemoteConfig.Addr = fmt.Sprintf("unix:%s", shared.VarPath("unix.socket"))
 		}
 	} else {
-		// Read the client certificate (if it exists)
-		clientCertPath := path.Join(config.ConfigDir, "client.crt")
-		if shared.PathExists(clientCertPath) {
-			certBytes, err := ioutil.ReadFile(clientCertPath)
-			if err != nil {
-				return nil, err
-			}
 
-			info.ClientPEMCert = string(certBytes)
+		certf := config.ConfigPath("client.crt")
+		keyf := config.ConfigPath("client.key")
+
+		if !shared.PathExists(certf) || !shared.PathExists(keyf) {
+			fmt.Fprintf(os.Stderr, i18n.G("Generating a client certificate. This may take a minute...")+"\n")
```

Comment:
Yes, I think we shouldn't be writing to `pwd` (or anywhere else on the filesystem) in NewClient, since it may be use by other golang binaries who want to manage their configuration differently. Can we do all the cert generation in lxc/* somewhere and then just use NewClientFromInfo?